src/service/dashboards/mod.rs

```diff
@@ -225,69 +189,57 @@ pub async fn move_dashboard(
-        } else if dashboard.version == 2 {
-            json::to_vec(&dashboard.v2.unwrap()).unwrap()
-        } else if dashboard.version == 3 {
-            json::to_vec(&dashboard.v3.unwrap()).unwrap()
-        } else if dashboard.version == 4 {
-            json::to_vec(&dashboard.v4.unwrap()).unwrap()
-        } else {
-            json::to_vec(&dashboard.v5.unwrap()).unwrap()
-        };
+    let Some(dashboard) = table::dashboards::get(org_id, from_folder, dashboard_id).await? else {
+        return Err(DashboardError::DashboadNotFound);
+    };
 
-        // add the dashboard to the destination folder
-        if let Err(error) =
-            db::dashboards::put(org_id, dashboard_id, to_folder, dash.into(), None).await
-        {
-            return Ok(Response::InternalServerError(error).into());
-        }
+    // make sure the destination folder exists
+    if !table::folders::exists(org_id, to_folder).await? {
+        return Err(DashboardError::MoveDestinationFolderNotFound);
+    };
 
-        // delete the dashboard from the source folder
-        let _ = db::dashboards::delete(org_id, dashboard_id, from_folder).await;
-        Ok(Response::OkMessage("Dashboard moved successfully".to_string()).into())
-    } else {
-        Ok(Response::NotFound("Dashboard".to_string()).into())
-    }
-}
+    // add the dashboard to the destination folder
+    put(org_id, dashboard_id, to_folder, dashboard, None).await?;
+
+    // delete the dashboard from the source folder
+    let _ = table::dashboards::delete(org_id, from_folder, dashboard_id).await;
 
-#[derive(Debug)]
-enum Response {
-    OkMessage(String),
-    NotFound(String),
-    Conflict(anyhow::Error),
-    InternalServerError(anyhow::Error),
+    // Ok(Response::OkMessage("Dashboard moved successfully".to_string()).into())
+    Ok(())
 }
 
-impl From<Response> for HttpResponse {
-    fn from(resp: Response) -> Self {
-        match resp {
-            Response::OkMessage(message) => Self::Ok().json(MetaHttpResponse::message(
-                http::StatusCode::OK.into(),
-                message,
-            )),
-            Response::NotFound(entity) => Self::NotFound().json(MetaHttpResponse::error(
-                http::StatusCode::NOT_FOUND.into(),
-                format!("{entity} not found"),
-            )),
-            Response::Conflict(err) => Self::Conflict().json(MetaHttpResponse::error(
-                http::StatusCode::CONFLICT.into(),
-                err.to_string(),
-            )),
-            Response::InternalServerError(err) => {
-                Self::InternalServerError().json(MetaHttpResponse::error(
-                    http::StatusCode::INTERNAL_SERVER_ERROR.into(),
-                    err.to_string(),
-                ))
-            }
+#[tracing::instrument(skip(dashboard))]
+async fn put(
+    org_id: &str,
+    dashboard_id: &str,
+    folder_id: &str,
+    mut dashboard: Dashboard,
+    hash: Option<&str>,
+) -> Result<Dashboard, DashboardError> {
```

Comment:
This function used to exist in `service/db/dashboards` but since that module didn't contain any logic other than this method, I've just moved that method into this module and removed the `service/db/dashboards` module